src/Loader.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pest\Plugin;
+
+use JsonException;
+
+/**
+ * @internal
+ */
+final class Loader
+{
+    /**
+     * determines if the plugin cache file was loaded.
+     *
+     * @var bool
+     */
+    private static $loaded = false;
+
+    /**
+     * holds the list of cached plugin instances.
+     *
+     * @var array<int, object>
```

Comment:
This object doesn't have any base interface? If we are following segregation principle maybe we should have an empty interface like `PestPlugin`.